### ISSUE TITLE
Test with Boost 1.66 on Windows

### ISSUE THREAD

**Issue by chwarr** [microsoft/bond#856]
Boost 1.66 was added to the CI matrix on Linux in commit e96a8deb44002d1c0e5092d618ac17d5e19c6dfc. We need to add it for Windows as well.

This is currently blocked on PR https://github.com/Microsoft/bond/pull/854.

**Comment by chwarr** [comment]
Boost 1.65 is outside of the range of Boost that Bond supports now. Closing this in favor of https://github.com/microsoft/bond/issues/855
